mailbox/cassandra/src/main/java/org/apache/james/mailbox/cassandra/mail/task/SolveMailboxInconsistenciesService.java

```diff
@@ -0,0 +1,455 @@
+        @Override
+        public Mono<Result> fix(Context context, CassandraMailboxDAO mailboxDAO, CassandraMailboxPathV2DAO pathV2DAO) {
+            return Mono.just(Result.COMPLETED);
+        }
+
+        @Override
+        public Mono<Boolean> isStillPertinent(Context context, CassandraMailboxDAO mailboxDAO, CassandraMailboxPathV2DAO pathV2DAO) {
+            return Mono.just(INCONSISTENCY_STILL_PRESENT);
+        }
+    };
+
+    /**
+     * The Mailbox is referenced in MailboxDAO but the corresponding
+     * reference is missing in MailboxPathDao.
+     *
+     * In order to solve this inconsistency, we can simply re-reference the mailboxPath.
+     */
+    private static class OrphanMailboxDAOEntry implements Inconsistency {
+        private final Mailbox mailbox;
+
+        private OrphanMailboxDAOEntry(Mailbox mailbox) {
+            this.mailbox = mailbox;
+        }
+
+        @Override
+        public Mono<Result> fix(Context context, CassandraMailboxDAO mailboxDAO, CassandraMailboxPathV2DAO pathV2DAO) {
+            return pathV2DAO.save(mailbox.generateAssociatedPath(), (CassandraId) mailbox.getMailboxId())
+                .map(success -> {
+                    if (success) {
+                        LOGGER.info("Inconsistency fixed for orphan mailbox {} - {}",
+                            mailbox.getMailboxId().serialize(),
+                            mailbox.generateAssociatedPath().asString());
+                        context.fixedInconsistencies.incrementAndGet();
+                        return Result.COMPLETED;
+                    } else {
+                        context.errors.incrementAndGet();
+                        LOGGER.warn("Failed fixing inconsistency for orphan mailbox {} - {}",
+                            mailbox.getMailboxId().serialize(),
+                            mailbox.generateAssociatedPath().asString());
+                        return Result.PARTIAL;
+                    }
+                });
+        }
+
+        /**
+         * We validate:
+         *  - That the mailbox record stil exists and that its path did not change
+         *  - That no corresponding path record exist
+         */
+        @Override
+        public Mono<Boolean> isStillPertinent(Context context, CassandraMailboxDAO mailboxDAO, CassandraMailboxPathV2DAO pathV2DAO) {
+            return mailboxDAO.retrieveMailbox((CassandraId) mailbox.getMailboxId())
+                .filter(currentMailbox -> currentMailbox.generateAssociatedPath().equals(mailbox.generateAssociatedPath()))
+                .flatMap(currentMailbox -> pathV2DAO.retrieveId(currentMailbox.generateAssociatedPath())
+                    .map(entry -> !INCONSISTENCY_STILL_PRESENT)
+                    .switchIfEmpty(Mono.just(INCONSISTENCY_STILL_PRESENT)))
+                .filter(stillPresent -> stillPresent)
+                .switchIfEmpty(Mono.fromCallable(() -> {
+                    context.errors.incrementAndGet();
```

Comment:
Demeter law

Reply:
Can you please explain me what that is?

Reply:
https://hackernoon.com/object-oriented-tricks-2-law-of-demeter-4ecc9becad85

`Law of Demeter(LoD) is more of a guideline than a principle to help reduce coupling between components`

I'm within the same component, not sure this should apply...

Reply:
It depends what you call a component.

You have a very big class, with nested classes, we should not couple things more than necessary.

`context.incrementErrors()` vs `context.errors.incrementAndGet()` is focusing on the what instead of how, which is always preferable.

Reply:
Then let's add 20 more lines if you wish.